framework/web/ErrorHandler.php

```diff
@@ -99,7 +99,7 @@ protected function renderException($exception)
                 $response->data = $result;
             }
         } elseif ($response->format === Response::FORMAT_HTML) {
-            if (YII_ENV_TEST || isset($_SERVER['HTTP_X_REQUESTED_WITH']) && $_SERVER['HTTP_X_REQUESTED_WITH'] === 'XMLHttpRequest') {
+            if (YII_ENV_TEST || Yii::$app->request->isAjax) {
```

Comment:
One possible implication is about fatal errors with not enough memory... these could fail in the case. Not sure about it though. Needs testing.